tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -11096,6 +11224,605 @@ TEST_F(ExecutionTest, PackUnpackTest) {
+  CComPtr<ID3D12Device> pDevice;
+  if (!m_ExperimentalModeEnabled  && !CreateDevice(&pDevice, D3D_SHADER_MODEL_6_9)) {
+    // TODOLONGVEC: This should fail. HLK requirements should select the test.
+    // But for exectests we can just return and skip. Will need to add a
+    // mechanism to faciliate that behavior in a good way that enforces it to
+    // always be on in OS builds.
+    WEX::Logging::Log::Comment("Device does not support SM 6.9. Can't run these tests.");
+    WEX::Logging::Log::Result(WEX::Logging::TestResults::Skipped);
+    return;
+  }
+
+  // TableParameterHandler is a helper that parses the xml file linked
+  // in the taef metadata for this test method. So it uses
+  // ShaderOpArithTable.xml. And more specifically, it looks for the table ID
+  // LongVectorTestTable_ScalarTests as defined in the TAEF boilerplate for this
+  // test method.
+  const int tableSize = sizeof(LongVectorBinaryOpParameters) / sizeof(TableParameter);
+  TableParameterHandler handler(LongVectorBinaryOpParameters, tableSize);
+ 
+  CW2A Target(handler.GetTableParamByName(L"ShaderOp.Target")->m_str);
+  CW2A Text(handler.GetTableParamByName(L"ShaderOp.Text")->m_str);
+
+  // Note: Each element in these vectors is an array of the values
+  // inside each <value> tag in ShaderOpArithTable.xml for this tests
+  // row. Multiple value tags mean a longer vector.
+  std::vector<WEX::Common::String> *Validation_VecInput1 =
+  &(handler.GetTableParamByName(L"Validation.VecInput1")->m_StringTable);
+  auto vecInput1 = parseStringsToNumbers<T>(Validation_VecInput1);
+
+  std::vector<WEX::Common::String> *Validation_VecInput2 =
+  &(handler.GetTableParamByName(L"Validation.VecInput2")->m_StringTable);
+  auto vecInput2 = parseStringsToNumbers<T>(Validation_VecInput2);
+
+  std::vector<WEX::Common::String> *Validation_VecExpected =
+  &(handler.GetTableParamByName(L"Validation.VecExpected")->m_StringTable);
+  auto vecExpected = parseStringsToNumbers<T>(Validation_VecExpected);
+
+  std::vector<WEX::Common::String> *Scalar_Inputs =
+  &handler.GetTableParamByName(L"Validation.ScalarInput")->m_StringTable;
+  auto scalarInputs = parseStringsToNumbers<T>(Scalar_Inputs);
+
+  // double is plenty of room for tolerance range
+  const double Validation_Tolerance =
+  handler.GetTableParamByName(L"Validation.Tolerance")->m_double;
+
+  const size_t inputVectorCount = vecInput1.size();
+  const bool hasScalarInput = scalarInputs.size() > 0;
+  const bool hasVecInput2 = vecInput2.size() > 0;
+  VERIFY_IS_FALSE(hasScalarInput == hasVecInput2, L"Expecting either a vector and vector op or a vector and scalar op");
+
+  // The two operand vectors must be the same size.
+  VERIFY_IS_TRUE(inputVectorCount == vecInput2.size() || inputVectorCount == scalarInputs.size());
+
+  // Additionally all elements (also vectors) in the vector must be the same
+  // size.
+  // TODO: Add check.
+  // Keep filling the vectors with 'copies' of themselves until we hit N
```

Comment:
Why?  Is this trying to make up for potential bugs in the XML?

Reply:
Yah, this is another reason I'm leaning towards not defining things in the XML. Too easy to mess this up.